signature/src/signatures/sigstore.rs

```diff
@@ -0,0 +1,254 @@
+    let mut merged_config = SigstoreConfig::default();
+
+    for entry in fs::read_dir(dir)? {
+        let entry = entry?;
+        let path = entry.path();
+        if path.is_dir() || path.extension() != Some(OsStr::new("yaml")) {
+            continue;
+        }
+        let path_str = path
+            .to_str()
+            .ok_or(anyhow!("Unkown error: path parsed failed!"))?;
+        let config_yaml_string = fs::read_to_string(path_str)?;
+        let config = serde_yaml::from_str::<SigstoreConfig>(&config_yaml_string)?;
+
+        // The "default-docker" only allowed to be defined in one config file.
+        if config.default_docker.is_some() {
+            if merged_config.default_docker.is_some() {
+                return Err(anyhow!(
+                    "Error parsing sigstore config: \"default-docker\" defined repeatedly!"
+                ));
+            }
+            merged_config.default_docker = config.default_docker;
+        }
+
+        // An image namespace is not allowed appear in two different configuration files.
+        if let Some(docker_config_map) = config.docker {
+            for (ns_name, ns_config) in docker_config_map.iter() {
+                if merged_config.is_namespace_exit(ns_name) {
+                    return Err(anyhow!(format!(
+                        "Error parsing sigstore config: {} defined repeatedly!",
+                        &ns_name
+                    )));
+                }
+
+                merged_config.inser_a_docker_config_entry(ns_name, ns_config);
+            }
+        }
+    }
+
+    Ok(merged_config)
+}
+
+// returns an URL string (sigstore base url) configured in sigstore config for the image.
+pub fn get_sigstore_base_url(image_ref: &Reference) -> Result<String> {
+    let sigstore_config = load_sigstore_config(REGISTRY_CONFIG_DIR)?;
+
+    if let Some(docker_config_map) = sigstore_config.docker {
+        // Look for a full match.
+        let image_id = image::get_image_identity(image_ref);
+        // Look for a match of the possible parent namespaces.
+        let mut image_ns = image::get_image_namespace(image_ref);
+
+        image_ns.insert(0, image_id);
+        for ns in image_ns.iter() {
+            if let Some(base_url) = docker_config_map.get(ns) {
+                return Ok(base_url.sigstore.to_string());
+            }
+        }
+    }
+
+    // Look for a default location
+    if let Some(default_config) = sigstore_config.default_docker {
+        return Ok(default_config.sigstore);
+    }
+
+    Err(anyhow!(
+        "There is no config info for this image in sigstore config files!"
+    ))
+}
+
+pub fn get_sigs_from_lookaside(sigstore: String) -> Result<Vec<Vec<u8>>> {
+    let mut res: Vec<Vec<u8>> = Vec::new();
+
+    // FIXME: Now only support get signatures from local files.
+    // only the url begins with "file://" can be parsed.
+    if sigstore[0..4] == *"file" {
+        let sigstore_dir_path = &sigstore[7..];
```

Comment:
**Bug**: How do you know the string is this many bytes long? If the string is shorter, this will panic and kill the application I'd strongly suggest not parsing URI's ourselves but instead using a crate to handle it.